compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -1772,6 +1772,59 @@ unsafe extern "C" {
         Scope: &'ll Metadata,
         InlinedAt: Option<&'ll Metadata>,
     ) -> &'ll Metadata;
+
+    pub(crate) fn LLVMDIBuilderCreateSubroutineType<'ll>(
+        Builder: &DIBuilder<'ll>,
+        File: Option<&'ll Metadata>, // (unused)
+        ParameterTypes: *const Option<&'ll Metadata>,
+        NumParameterTypes: c_uint,
+        Flags: DIFlags, // (optional; default is `DIFlags::FlagZero`)
+    ) -> &'ll Metadata;
+
+    pub(crate) fn LLVMDIBuilderCreateUnionType<'ll>(
```

Comment:
same comment except for Name and NameLen, Elements and ElementsLen, UniqueId and UniqueIdLen, also please note whether the obvious should be a CStr with or _without_ its nul byte